mir_eval/separation.py

```diff
@@ -161,28 +166,139 @@ def bss_eval_sources(reference_sources, estimated_sources):
+            for jtrue in range(nsrc):
+                s_true, e_spat, e_interf, e_artif = \
+                    _bss_decomp_mtifilt(reference_sources,
+                                        estimated_sources[jest],
+                                        jtrue, 512)
+                sdr[jest, jtrue], sir[jest, jtrue], sar[jest, jtrue] = \
+                    _bss_source_crit(s_true, e_spat, e_interf, e_artif)
+    else:
+        # compute criteria for only the simple correspondence
+        # (estimate 1 is estimate corresponding to reference source 1, etc.)
+        sdr = np.empty(nsrc)
+        sir = np.empty(nsrc)
+        sar = np.empty(nsrc)
+        for j in range(nsrc):
             s_true, e_spat, e_interf, e_artif = \
                 _bss_decomp_mtifilt(reference_sources,
-                                    estimated_sources[jest],
-                                    jtrue, 512)
-            sdr[jest, jtrue], sir[jest, jtrue], sar[jest, jtrue] = \
+                                    estimated_sources[j],
+                                    j, 512)
+            sdr[j], sir[j], sar[j] = \
                 _bss_source_crit(s_true, e_spat, e_interf, e_artif)
 
-    # select the best ordering
-    perms = list(itertools.permutations(range(nsrc)))
-    mean_sir = np.empty(len(perms))
-    dum = np.arange(nsrc)
-    for (i, perm) in enumerate(perms):
-        mean_sir[i] = np.mean(sir[perm, dum])
-    popt = perms[np.argmax(mean_sir)]
-    idx = (popt, dum)
-    return (sdr[idx], sir[idx], sar[idx], popt)
+    # does user desire permutations?
+    if compute_permutation:
+        # select the best ordering
+        perms = list(itertools.permutations(list(range(nsrc))))
+        mean_sir = np.empty(len(perms))
+        dum = np.arange(nsrc)
+        for (i, perm) in enumerate(perms):
+            mean_sir[i] = np.mean(sir[perm, dum])
+        popt = perms[np.argmax(mean_sir)]
+        idx = (popt, dum)
+        return (sdr[idx], sir[idx], sar[idx], popt)
+    else:
+        # return the default permutation for compatibility
+        popt = range(nsrc)
+        return (sdr, sir, sar, popt)
+
+
+def bss_eval_sources_framewise(
+    reference_sources,
+    estimated_sources,
+    win,
+    hop,
+    compute_permutation=False
+):
+    """Framewise computation of bss_eval_sources
+
+    Examples
+    --------
+    >>> # reference_sources[n] should be an ndarray of samples of the
+    >>> # n'th reference source
+    >>> # estimated_sources[n] should be the same for the n'th estimated
+    >>> # source
+    >>> (sdr, sir, sar,
+    ...  perm) = mir_eval.separation.bss_eval_sources_framewise(
+             reference_sources,
+    ...      estimated_sources)
+
+    Parameters
+    ----------
+    reference_sources : np.ndarray, shape=(nsrc, nsampl)
+        matrix containing true sources
+    estimated_sources : np.ndarray, shape=(nsrc, nsampl)
+        matrix containing estimated sources
+    win : int, optional
```

Comment:
It is required that the number of sources and samples be the same. I'll add a commenting stating that to the functions.